appscale/tools/appscale_stats.py

```diff
@@ -0,0 +1,655 @@
+  ]
+
+  for proc in raw_process_stats.itervalues():
+    process_list += proc["processes_stats"]
+
+  for proc in process_list:
+    name = proc["unified_service_name"]
+    app_id = proc["application_id"]
+    name_id = name + (" ({})".format(app_id) if app_id else "")
+    if name_id in unique_processes:
+      # calculation with process has been made
+      continue
+
+    process_group = [
+      p for p in process_list
+      if p["unified_service_name"] == name
+      and p["application_id"] == app_id
+    ]
+
+    unique_processes.add(name_id)
+    unique_memory = 0
+    cpu_percent = 0
+
+    for p in process_group:
+      unique_memory += int(p["memory"]["unique"])
+      cpu_percent += float(p["cpu"]["percent"])
+      if p["children_num"] > 0:
+        child = p["children_stats_sum"]
+        unique_memory += int(child["memory"]["unique"])
+        cpu_percent += float(child["cpu"]["percent"])
+
+    avg_cpu_per_process = cpu_percent / len(process_group)
+    cpu_percent_cpu_count = cpu_percent / cpu_count
+
+    process_stats.append([
+      name_id,
+      len(process_group),
+      unique_memory / 1024 / 1024,  # row[2] from B to MB
+      cpu_percent,
+      avg_cpu_per_process,
+      cpu_percent_cpu_count
+    ])
+
+  return process_stats_headers, process_stats
+
+
+def get_proxy_stats_rows(raw_proxy_stats, verbose, apps_filter):
+  """
+  Obtains useful information from proxy statistics and returns:
+  SERVICE (ID), UNIQUE MEMORY SUM (MB), CPU PER 1 PROCESS (%),
+  CPU PER 1 CORE (%), CPU SUM (%) values.
+
+  Args:
+    raw_proxy_stats: A dict in which each key is an ip and value is a dict
+      of useful information.
+    verbose: A boolean - verbose or not verbose mode.
+    apps_filter: A boolean - show all services or applications only.
+
+  Returns:
+    A list of proxy statistics headers.
+    A list of proxy statistics.
+  """
+  if verbose:
+    headers = [
+      "SERVICE (ID)", "SERVERS | DOWN", "RATE | REQ TOTAL", "5xx | 4xx",
+      "BYTES IN | BYTES OUT", "QCUR | SCUR", "QTIME | RTIME"
+    ]
+  else:
+    headers = [
+      "SERVICE (ID)", "SERVERS | DOWN", "RATE | REQ TOTAL", "5xx | 4xx", "QCUR"
+    ]
+
+  proxy_stats = []
+  proxy_groups = []
+  unique_proxies = {}
+
+  for node in raw_proxy_stats.itervalues():
+    proxy_groups += node["proxies_stats"]
+
+  for node in proxy_groups:
+    if node["application_id"]:
+      service_name_id = "app ({app_id})".format(app_id=node["application_id"])
+    elif not apps_filter:
+      service_name_id = node["unified_service_name"]
+    else:
+      continue
+
+    if service_name_id not in unique_proxies:
+      summary_proxy = unique_proxies[service_name_id] = defaultdict(int)
+    else:
+      summary_proxy = unique_proxies[service_name_id]
+
+    summary_proxy["servers_count"] = node["servers_count"]
+    servers_down = sum(1 for s in node["servers"] if s["status"] != "UP")
+    summary_proxy["servers_down_count"] = servers_down
```

Comment:
Combine line 564 + 563. 
summary_proxy["servers_down_count"] = sum(1 for s in node["servers"] if s["status"] != "UP")

Reply:
@scragraham the combination of two lines has 96 signs which exceeds our agreement (80 signs).
I have doubts if splitting it as 
```
    summary_proxy["servers_down_count"] = sum(
      1 for s in node["servers"] if s["status"] != "UP")
```
or other ways without introducing intermediate variable would have better readability.

Reply:
you can wrap the if stmt of the comprehension:
```
    summary_proxy["servers_down_count"] = sum(1 for s in node["servers"]
                                              if s["status"] != "UP")

````
Also: the world won't come to an end if you exceed 80 characters (I think)